File: PHP/ZigZagSequence.php
<?php
/**
 * @version PHP 7.3.29
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class ZigZagSequence {

    /*
     * Complete the 'findZigZagSequence' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */
    function findZigZagSequence($arr) {
        //inicializa var de retorno
        $res = array();

        //identifica a metade do array, -1 pq o array inicia em 0.
        $k = ((count($arr)+1)/2)-1;

        //identifica o menor valor do array, q inicia o retorno.
        $menor = min($arr);

        //monta resultado.
        array_search($menor, $arr) > $k 
            ? $res = array_merge( array_slice($arr, array_search($menor, $arr), $k), array_reverse(array_slice($arr, 0, $k+1)) ) 
            : $res = array_merge( array_slice($arr, 0, $k), array_reverse(array_slice($arr, $k, $k+1)) );

        return $res;
    }

    /*
    $_fp = fopen(getenv("OUTPUT_PATH"), "w");
    $t = intval(trim(fgets(STDIN)));
    $n = intval(trim(fgets(STDIN)));
    $arr_temp = rtrim(fgets(STDIN));
    $arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

    $result = findZigZagSequence($arr);
    fwrite($_fp, implode(" ", $result) . "\n");
    fclose($_fp);
    */
}

File: JAVA/ZigZagSequence.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.IntStream;

public class ZigZagSequence {

    //private static final String ArrayUtils = null;

    public static void main (String[] args) throws java.lang.Exception {
        try (Scanner kb = new Scanner(System.in)) {
            int test_cases = kb.nextInt();
            for(int cs = 1; cs <= test_cases; cs++){
                int n = kb.nextInt();
                 int a[] = new int[n];
                for(int i = 0; i < n; i++){
                    a[i] = kb.nextInt();
                }
                findZigZagSequence(a, n);
            }
        }
   }

    public static int[] findZigZagSequence(int[] arr, int n) {
        //inicializa var de retorno
        int[] res = new int[n];
    
        //identifica a metade do array, -1 pq o array inicia em 0.
        int k = ((arr.length+1)/2)-1;
    
        //identifica o menor valor do array, q inicia o retorno.
        int menor = Arrays.stream(arr).min().getAsInt();
    
        //monta resultado.
        int pos = indexOfIntArray( arr, menor );
        if ( pos > k ) {
            int[] parte1 = Arrays.copyOfRange( arr, pos, arr.length );
            int[] parte2 = Arrays.copyOfRange( arr, 0, k+1 );
            reverse(parte2);

            res = IntStream.concat( Arrays.stream(parte1), Arrays.stream(parte2) ).toArray();
        } else {
            int[] parte1 = Arrays.copyOfRange( arr, 0, k );
            int[] parte2 = Arrays.copyOfRange( arr, k, n );
            reverse(parte2);

            res = IntStream.concat( Arrays.stream(parte1), Arrays.stream(parte2) ).toArray();
        }
    
        //System.out.println(Arrays.toString(res).replaceAll(",", " ").replaceFirst("]","").replaceFirst("[", ""));
        return res;
    }

    private static int indexOfIntArray(int[] array, int key) {
        int returnvalue = -1;
        for (int i = 0; i < array.length; ++i) {
            if (key == array[i]) {
                returnvalue = i;
                break;
            }
        }
        return returnvalue;
    }

    private static void reverse(int a[]) {
        int i, t;
        int n = a.length;
        for (i = 0; i < n / 2; i++) {
            t = a[i];
            a[i] = a[n - i - 1];
            a[n - i - 1] = t;
        }
    }
}
